torch/_dynamo/variables/tensor.py

```diff
@@ -707,18 +711,57 @@ def var_getattr(self, tx, name):
             result = wrap_fx_proxy_cls(
                 target_cls=NumpyNdarrayVariable,
                 tx=tx,
-                proxy=GetAttrVariable.create_getattr_proxy(self.as_proxy(), name),
+                proxy=tx.output.create_proxy(
+                    "call_function",
+                    attr_wrapper,
+                    (self.as_proxy(), name),
+                    {},
+                ),
                 example_value=None,
                 **options,
             )
-        elif name in ["size", "itemsize", "strides", "shape", "ndim"]:
+        elif name in ["ndim", "itemsize", "shape"]:
             result = wrap_fx_proxy_cls(
                 target_cls=ConstantVariable,
                 tx=tx,
                 proxy=GetAttrVariable.create_getattr_proxy(self.as_proxy(), name),
                 example_value=None,
                 **options,
             )
+        elif name == "shape":
+            result = wrap_fx_proxy_cls(
+                target_cls=TupleVariable,
+                tx=tx,
+                proxy=GetAttrVariable.create_getattr_proxy(self.as_proxy(), name),
+                example_value=None,
+                **options,
+            )
```

Comment:
So, if I understand correctly, you're taking advantage of the fact that Tensor also has an equivalent shape attribute to turn this into a direct attr access. But you also needed to override the target_cls because PyTorch returns torch.Size while numpy returns tuple. This all seems worth commenting about in the source code.